### PR TITLE
WT-9048 Sweep tiered dhandle as soon as its metadata gets removed.

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**

### CODE REVIEW

**Comment by AlexeyAnisimovAU** [comment]
Evergreen failures are not related to this change.

**Review by ddanderson — COMMENTED** [review]
My main concern here is that both the sweep server and the application thread may be walking the list at exactly the same time and both be trying to discard the dhandle. I see that the remove code has write locks, so they will try to remove sequentially.  But I think that both could be holding a dhandle pointer, and one finishes removing, the dhandle memory will be freed - leading to a bad reference by the other thread when it calls `__sweep_remove_one`.

Perhaps one solution would be instead of forcibly removing on the application side, maybe something else that indicates it shouldn't be used.  I was thinking about zeroing out the dhandle->name field, but I think that's assumed to be non-null in a bunch of places.

What happens elsewhere when a created dhandle needs to be "rolled back"?  Is it the fact that we are creating multiple dhandles as part of the tiered open that makes this case different?  Or is it different?

File: src/conn/conn_sweep.c [hunk 1]
```diff
@@ -206,24 +206,29 @@ __sweep_remove_one(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
...
     conn = S2C(session);
+    uri_len = (uri == NULL ? 0 : strlen(uri));
 
     TAILQ_FOREACH_SAFE(dhandle, &conn->dhqh, q, dhandle_tmp)
     {
         if (WT_IS_METADATA(dhandle))
             continue;
         if (!WT_DHANDLE_CAN_DISCARD(dhandle))
             continue;
+        if (uri != NULL && !WT_STRING_MATCH(uri, dhandle->name, uri_len))
```
[the remark this answers is not in the document]
Comment (AlexeyAnisimovAU) [reply src/conn/conn_sweep.c]: This should fix the string comparison issue [ab8a888](https://github.com/wiredtiger/wiredtiger/pull/7968/commits/ab8a888cafc70d37d92e008b2ac4f7392655772a)
I need to think more about the comment about the possible collision with the sweep server over the dhandles list.

**Comment by AlexeyAnisimovAU** [comment]
Closing this PR since the proper fix has been implemented: https://github.com/wiredtiger/wiredtiger/pull/7987

**Pull request closed without merging**